src/index.ts

```diff
@@ -378,6 +380,63 @@ class MayanRouteBase<N extends Network> extends routes.AutomaticRoute<
     return instructions;
   }
 
+  async injectReferralInstructionsForSui(
+    provider: SuiClient,
+    request: routes.RouteTransferRequest<N>,
+    sender: string,
+    originalAmount: string,
+  ) {
+    const { fromChain, source } = request;
+    const referralParams = this.getReferralParameters(request);
+    const { isNewSuiReferralEnabled } = referralParams;
+    const referrerAddress = this.referrerAddress()?.sui;
+    const referralFee = this.getFeeInBaseUnits(request, originalAmount);
+
+    if (
+      !referrerAddress ||
+      !referralFee ||
+      fromChain.network !== 'Mainnet' ||
+      !isNewSuiReferralEnabled
+    ) {
+      return {};
+    }
+
+    const tx = new Transaction();
+    const token = source.id.address;
+    const isSui = isNative(token);
+    let primaryCoinObjectId;
+
+    if (isSui) {
+      primaryCoinObjectId = tx.gas;
+    } else {
+      const coinType = token.toString();
+
+      const [primaryCoin] = await SuiPlatform.getCoins(
+        provider,
+        sender,
+        coinType,
+      );
+
+      if (!primaryCoin) {
+        throw new Error(`No coins of type ${coinType} found for ${sender}.`);
+      }
+
+      primaryCoinObjectId = tx.object(primaryCoin.coinObjectId);
+
+      // if (mergeCoins.length > 0) {
```

Comment:
Adding this part back in gives me an error in Phantom wallet "Failed to fetch network fee"